src/helper.jl

```diff
@@ -53,21 +53,30 @@ Base.unsafe_convert(T::Type{Ptr{Nothing}}, t::CWrapper) = (t.ptr == C_NULL) ? th
 mutable struct XpressProblem <: CWrapper
     ptr::Lib.XPRSprob
     callback::Vector{Any}
+    finalize_env::Bool
     time::Float64
     logfile::String
-    function XpressProblem(; logfile = "")
-        ref = Ref{Lib.XPRSprob}()
-        createprob(ref)
-        @assert ref[] != C_NULL "Failed to create XpressProblem. Received null pointer from Xpress C interface."
-        p = new(ref[], Any[], 0.0, logfile)
-        if logfile != ""
-            setlogfile(p, logfile)
+    function XpressProblem(ptr::Lib.XPRSprob; finalize_env::Bool = true, logfile = "")
+        model = new(ptr, Any[], finalize_env, 0.0, logfile)
+        if finalize_env
+            finalize(() -> destroyprob(model))
```

Comment:
See https://github.com/jump-dev/Xpress.jl/pull/100/commits/93d3ecec271315aae60d7e3cfa07e6effd62c450 for correct use of `finalizer`, the old code was incorrect.